### ISSUE TITLE
Sidebar: forms can't be submitted in the presence of non-modal Sidebar

### ISSUE THREAD

**Issue by inad9300** [primefaces/primereact#3360]
### Describe the bug

Likely related to https://github.com/primefaces/primereact/issues/2562

### Reproducer

https://codesandbox.io/s/primereact-test-forked-ghjz97?file=/src/index.js

### PrimeReact version

8.6.0

### React version

18.x

### Language

ALL

### Build / Runtime

Create React App (CRA)

### Steps to reproduce the behavior

1. Click "Submit"
2. Verify that no message is logged

**Comment by melloware** [comment]
I agree it looks like when modal={false} the mask is still filling the screen and intercepting clicks.

**Comment by atrhacker** [comment]
Seems to be linked to the following: 
https://forum.primefaces.org/viewtopic.php?f=57&t=74993

Basically, if you put any input type file (or FileUpload) in the page when there is a sidebar, the behavior of the input file will be completely blocked. (with or without form attached to it)
You can test it here:
https://codesandbox.io/s/withered-water ... oadDemo.js](https://codesandbox.io/s/withered-waterfall-cs307o?file=/src/demo/FileUploadDemo.js

Not sure, if I need to open a different bug request for this or shall we attach it to this one?

**Comment by melloware** [comment]
I think this one is fine it's most likely the same issue

**Comment by kl-nevermore** [comment]
@atrhacker 
you can change `onClose` as
```tsx
const onClose = (event) => {
        props.onHide();
       props.modal && event.preventDefault();
    };
```

**Comment by kl-nevermore** [comment]
if modal is false, should the Sidebar be closed and preventDefault when clicked outside? 
Or need a new props to decide and the new props respect `props.modal`?
I checked Mui and Antd they both can support
Chakra-ui is not supported because the root container width is 100vw


Like this scene, not yet possible
![20240104175032_rec_](https://github.com/primefaces/primereact/assets/25167742/89aa2a5e-d460-450f-93e5-7edf57960f7b)
